components/automate-cli/cmd/chef-automate/start_linx.go

```diff
@@ -9,59 +10,293 @@ import (
+	Error  error
+}
+
+var startCmdFlags = struct {
+	automate   bool
+	chefServer bool
+	opensearch bool
+	postgresql bool
+}{}
+
 var startCommand = &cobra.Command{
 	Use:   "start",
 	Short: "Start Chef Automate",
 	RunE:  runStartCmd,
 	Annotations: map[string]string{
-		docs.Tag: docs.FrontEnd,
+		docs.Tag: docs.BastionHost,
 	},
 }
 
+func init() {
+	startCommand.PersistentFlags().BoolVarP(&startCmdFlags.automate, "automate", "a", false, "start chef automate service on automate nodes")
+	startCommand.PersistentFlags().BoolVar(&startCmdFlags.automate, "a2", false, "start chef automate service on automate nodes[DUPLICATE]")
+	startCommand.PersistentFlags().BoolVarP(&startCmdFlags.chefServer, "chef_server", "c", false, "start chef automate service on chef-server nodes")
+	startCommand.PersistentFlags().BoolVar(&startCmdFlags.chefServer, "cs", false, "start chef automate service on chef-server nodes[DUPLICATE]")
+	startCommand.PersistentFlags().BoolVarP(&startCmdFlags.opensearch, "opensearch", "o", false, "start hab-sup service on opensearch nodes")
+	startCommand.PersistentFlags().BoolVar(&startCmdFlags.opensearch, "os", false, "start hab-sup service on opensearch nodes[DUPLICATE]")
+	startCommand.PersistentFlags().BoolVarP(&startCmdFlags.postgresql, "postgresql", "p", false, "start hab-sup service on postgresql nodes")
+	startCommand.PersistentFlags().BoolVar(&startCmdFlags.postgresql, "pg", false, "start hab-sup service on postgresql nodes[DUPLICATE]")
+	RootCmd.AddCommand(startCommand)
+}
+
 func runStartCmd(cmd *cobra.Command, args []string) error {
-	writer.Title("Starting Chef Automate")
-	if isDevMode() {
-		if err := exec.Command("hab", "sup", "status").Run(); err == nil {
-			return nil
+	if isA2HARBFileExist() {
+		if !startCmdFlags.automate && !startCmdFlags.chefServer && !startCmdFlags.opensearch && !startCmdFlags.postgresql {
+			writer.Println(cmd.UsageString())
+		}
+		infra, err := getAutomateHAInfraDetails()
+		if err != nil {
+			return err
+		}
+		if err = runStartCommandHA(infra, args); err != nil {
+			return err
+		}
+	} else if isDevMode() {
+		if err := runStartDevMode(); err != nil {
+			return err
 		}
 
-		if err := os.MkdirAll("/hab/sup/default", 0755); err != nil {
-			return status.Annotate(err, status.FileAccessError)
+	} else {
+		if err := runStartStandalone(); err != nil {
+			return err
 		}
-		writer.Title("Launching the Habitat Supervisor in the background...")
-		out, err := os.Create("/hab/sup/default/sup.log")
+	}
+	return nil
+}
+
+func runStartCommandHA(infra *AutomateHAInfraDetails, args []string) error {
+	sshConfig := &SSHConfig{
+		sshUser:    infra.Outputs.SSHUser.Value,
+		sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+		sshPort:    infra.Outputs.SSHPort.Value,
+	}
+	sshUtil := NewSSHUtil(sshConfig)
+	errorList := list.New()
+	var err error
+	if startCmdFlags.automate {
+		frontendIps := infra.Outputs.AutomatePrivateIps.Value
+		err = checkNodes(args, sshUtil, frontendIps, "automate", writer)
```

Comment:
Making checkNodes testable:
create a map of sshUtil here which should be map[string]SSHUtil
```suggestion
               var sshUtilMap map[string]SSHUtil
               sshUtilMap = createSSHUtilMap(frontendIps, sshConfig)
		err = checkNodes(args, sshUtil, frontendIps, "automate", writer)
```